extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -715,6 +716,21 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return {'artc': artc.join() || null, 'ati': ati.join(), 'ard': ard.join()};
   }
 
+  /** @override */
+  getCustomRealTimeConfigMacros_() {
+    const macros = {};
+    const docInfo = Services.documentInfoForDoc(this.element);
+    macros['PAGEVIEWID'] = docInfo.pageViewId;
```

Comment:
const docInfo = ..
return {
   'HREF': this.win.location.href,
   'DATASLOT': ...
};

Also I would have expected these to be function values such that they are only evaluated if/when the macro exists.  In fact, you could make it so that the element is passed to the function thereby allowing this to be a static object.  Seems much more efficient.

Reply:
changed to function values